libfreerdp/codec/color.c

```diff
@@ -3312,6 +3474,16 @@ int freerdp_image_copy(BYTE* pDstData, DWORD DstFormat, int nDstStep, int nXDst,
 		status = freerdp_image8_copy(pDstData, DstFormat, nDstStep, nXDst, nYDst,
 				nWidth, nHeight, pSrcData, SrcFormat, nSrcStep, nXSrc, nYSrc, palette);
 	}
+	else if (srcBitsPerPixel == 1)
```

Comment:
The mix of if with srcBitsPerPixel and srcBytesPerPixel is a little misleading.
What's the value of srcBytesPerPixel when we hit the case srcBitsPerPixel = [1 | 4] ? If it's 0 then perhaps we could add the appropriate if() and catch srcBitsPerPixel in that statement...

Note: we silently ignore srcBytesPerPixel not in [1, 2, 3, 4] we should at least warn when we hit these cases